### DISCUSSION TITLE
How to set log option in Wpf.ChromiumWebBrowser?

### DISCUSSION THREAD

**Discussion by tp1415926535** [cefsharp/CefSharp#3677] [Q&A] [answered]
I've know that there's a way to set by refer CefSettings class:
`            CefSettings settings = new CefSettings();         
            settings.LogSeverity = LogSeverity.Error;`
However I use this browser in WPF, so the setting is `browser.BrowserSettings` which belongs to IBrowserSettings class.
So how to set log option in WPF? I've tried to `Cef.Initialize(settings)` but it will conflict with which browser has been declared. I just want to add a log setting to it, thanks.

**Comment by amaitland** [comment, marked as answer]
> CefSettings settings = new CefSettings(); settings.LogSeverity = LogSeverity.Error;

That is the only way to specify the log options. 

> However I use this browser in WPF, so the setting is browser.BrowserSettings which belongs to IBrowserSettings class.

You will need to call Cef.Initialize in code, generally I'd suggest in your App class.

https://github.com/cefsharp/CefSharp.MinimalExample/blob/cefsharp/91/CefSharp.MinimalExample.Wpf/App.xaml.cs#L40

You cannot set log severity using IBrowserSettings.

**Reply by tp1415926535** [reply]
Well, I used `Cef.Initialize()` instead of defining the browser in xaml, and it seemed to work. But when I closed the browser window , called `Cef.Shutdown()` and then opened it a second time, the program crash without any error, can you tell me what this is about by the way?

**Reply by amaitland** [reply]
> Well, I used Cef.Initialize() instead of defining the browser in xaml, and it seemed to work.

You can still create your ChromiumWebBrowser instance in xaml. There is no need to create it in code to call Cef.Initialize. 

> called Cef.Shutdown() and then opened it a second time, the program crash without any error, can you tell me what this is about by the way?

It's rare that you need to call Cef.Shutdown manually, it is called on application exit. https://github.com/cefsharp/CefSharp/blob/cefsharp/91/CefSharp.Wpf/ChromiumWebBrowser.cs#L388

You would need to provide an example that reproduces the problem for me to know what's going on. 

I'd suggest testing with https://github.com/cefsharp/CefSharp.MinimalExample

**Reply by tp1415926535** [reply]
Oh, I get it, the only problem is that I put `Cef.Initialize`together with the InitializeComponent() event of the child window, instead of the initialization event of the whole program which in App.xaml.cs, so there are all sorts of errors being reported :)
